rllib/core/learner/learner_group.py

```diff
@@ -324,6 +359,38 @@ def _get_results(self, results):
                 raise result_or_error
         return processed_results
 
+    def _get_tagged_results(self, results):
+        """Get results from the worker manager and group them by tag.
+
+        Returns:
+            A list of lists of results, where each inner list contains all results
+            for same tags.
+
+        """
+        unprocessed_results = []
```

Comment:
Why not store the tagged results in a dict mapping from tag to list of results? This gives you O(1) on the read.

```
processed_results = defaultdict(list)
for result:
     if ok:
           processed_results[result.tag].append(result_or_error)
```
Then
```
for tag in processed_results:
      self._inflight_request_tags.remove(tag)
```

Reply:
I can make this change but for the given amount of tags << 10000, this isn't going to make any significant difference to actual runtime. Since we aren't operating on a large amount of data here, I'll rewrite this in a way where its as easy to modify/maintain as possible.

Reply:
all done!